docs/dependency-injection.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Dependency injection in Blazor
+author: rstropek
+description: Learn how Blazor apps can use built-in services by having them injected into components.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/09/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/dependency-injection
+---
+# Dependency injection in Blazor
+
+By [Rainer Stropek](https://www.timecockpit.com)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+Blazor has [dependency injection (DI)](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection) built-in. Blazor apps can use built-in services by having them injected into components. Blazor apps can also define custom services and make them available via DI.
+
+## What is dependency injection?
```

Comment:
Firstly, I really like the overall tone of voice here and think this is just the right amount of background detail. Great job!

Secondly, I *slightly* object to the idea that "component should *not* create an instance of `DataAccess`" as if that were always true. That's entirely the choice of the application developer. Sometimes there's value in working with an abstraction; sometimes it would just be unnecessary friction. It depends on the scenario. The DI system in Blazor is equally happy to supply instances of concrete service classes as it is to deal with interfaces.

I'd say something like:

DI is a technique for accessing services configured in a central location. This can be useful if:

* You want to share a single instance of a service class across many components (known as a "singleton" service)
* Or, if you want to decouple components from particular concrete service classes and only reference abstract interfaces. For example, you might have an interface `IDataAccess` implemented by a concrete class `DataAccess`. If your components use DI to receive an `IDataAccess` implementation, they aren't coupled to any concrete type. That means the implementation could easily be swapped, perhaps to a mock implementation in unit tests.

Blazor's DI system is responsible for supplying instances of services to components. It also resolves dependencies recursively, so that services themselves can depend on further services, and so on. DI is configured during startup of the app. An example is shown later in this topic.